src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(exemplar) {
+    if (!exemplar.hidden && exemplar.__proto__ !== Carnivore.prototype) {
+      exemplar.health -= 50;
+
+      if (exemplar.health) {
+        Animal.alive.splice(Animal.alive.indexOf(exemplar), 1);
```

Comment:
```suggestion
      if (exemplar.health) {
        Animal.alive.splice(Animal.alive.indexOf(exemplar), 1);
```
this part should work only if health === 0